### ISSUE TITLE
Replace bytes.Equal with subtle.ConstantTimeCompare

### ISSUE THREAD

**Issue by dlespiau** [jcmturner/gokrb5#215]
It's nice to use constant time comparison functions in crypto code so the time it takes for the comparison to fail cannot be used to deduce how many bytes were correct. The `crypto/subtle` package has exactly that! https://golang.org/pkg/crypto/subtle/#ConstantTimeCompare

**Comment by jcmturner** [comment]
Thanks for the tip!
I have looked through the code for bytes.Equal and I only see it being used for comparing computed checksums with the ones provided. Given a checksum cannot be used to deduce anything about the value over which it is computed I cannot see what could usefully be figured out from the time it takes to compare checksum values. I'd be interested to know if you agree with this or not.

**Comment by dlespiau** [comment]
First, just to be clear, I'm definitely not a crypto expert :) 

I noticed that the HMAC equal() takes great care of doing that: https://golang.org/src/crypto/hmac/hmac.go?s=2543:2577#L86. 

I also noticed that the C library uses constant time comparisons:
https://github.com/krb5/krb5/blob/master/src/lib/crypto/krb/verify_checksum.c#L75
https://github.com/krb5/krb5/blob/09c9b7d6f64767429e90ad11a529e6ffa9538043/src/include/k5-platform.h#L1074

So I looked for other comparisons in this library.

In the general case, there are definitely timing attacks in the wild using this. Now, is this cargo culting or it actually prevent an attack vector, I'm not so sure. It may depend on which checksum we are talking about:  the "internal" checksums, the ones appended to the ciphertext or the ones guaranteeing   message integrity. I haven't really deeply think about it.
